packages/analytics-js-integrations/src/integrations/CommandBar/browser.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable class-methods-use-this */
+import {
+  NAME,
+  DISPLAY_NAME,
+} from '@rudderstack/analytics-js-common/constants/integrations/CommandBar/constants';
+import Logger from '../../utils/logger';
+import { loadNativeSdk } from './nativeSdkLoader';
+
+const logger = new Logger(DISPLAY_NAME);
+
+class CommandBar {
+  constructor(config, analytics, destinationInfo) {
+    if (analytics.logLevel) {
+      logger.setLogLevel(analytics.logLevel);
+    }
+    this.analytics = analytics;
+    this.orgId = config.orgId;
+    this.name = NAME;
+    ({
+      shouldApplyDeviceModeTransformation: this.shouldApplyDeviceModeTransformation,
+      propagateEventsUntransformedOnError: this.propagateEventsUntransformedOnError,
+      destinationId: this.destinationId,
+    } = destinationInfo ?? {});
+  }
+
+  init() {
+    loadNativeSdk(this.orgId);
+  }
+
+  isLoaded() {
+    return window.CommandBar && typeof window.CommandBar.shareState === 'function';
+  }
+
+  isReady() {
+    return this.isLoaded();
+  }
+
+  /**
+   * Idenitfies a user with userId and traits
+   * Commandbar doc: https://www.commandbar.com/docs/sdk/lifecycle/boot/
+   * @param {*} rudderElement
+   * @returns
+   */
+  identify(rudderElement) {
+    const { message } = rudderElement;
+    const { traits, userId, context } = message;
+    const userTraits = { ...context.traits, ...traits };
+    const { hmacId } = userTraits;
+    if (!userId) {
+      logger.error('userId from identify call is missing');
+      return;
```

Comment:
will this actually drop the event? @MoumitaM to confirm

Reply:
It wont actually drop any event but yeah will not make any call to commandbar as before reaching that statement it will return

Reply:
Yes, it'll drop the event to CommandBar (CB) as we're not reaching the stage where data is forwarded to the CB SDK.